lib/rules/no-unsafe-window-use.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * @fileoverview Rule to avoid unsafe global window use
+ * @author Johnny Zabala
+ */
+
+'use strict';
+
+const docsUrl = require('../util/docsUrl');
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+function isMethod(node) {
+  return node.type === 'Property'
+    || node.type === 'MethodDefinition'
+    || node.type === 'ClassProperty';
+}
+
+function isAllowedLifecycle(node) {
+  return ['componentDidMount', 'componentDidUpdate', 'componentWillUnmount'].includes(node.key.name);
+}
+
+function isAllowedHook(node) {
+  return (node.type === 'CallExpression')
+  && ['useEffect', 'useLayoutEffect'].includes(node.callee.name);
```

Comment:
`.includes` won't work on node 4; use the `array-includes` package. or, for such short lists, just use `||` and `===` comparisons :-)

Reply:
Done!